### PR TITLE
Add examples for HANA self-query (Advanced RAG)

### PR DESCRIPTION
**Pull request by prashantrakheja**
This PR adds examples for HANA self-query under the 'Advanced RAG' section.

### CODE REVIEW

**Review by EvgeniiSkrebtcov — CHANGES_REQUESTED** [review]
As far as I can see, this example is not covered in the readme. Is it a reimplementation of a [Self-Query example](https://github.com/SAP-samples/btp-genai-starter-kit/blob/main/scripts/step03_explore_examples/02_advanced_rag/README.md#example-self-query) or something else?

File: scripts/step03_explore_examples/02_advanced_rag/main.py [hunk 1]
```diff
@@ -36,7 +37,8 @@ def main():
         print("3: Advanced RAG - Self Query")
         print("4: Advanced RAG - Rewrite Retrieve Read")
         print("5: Advanced RAG - RAG Fusion")
-        print("6: Exit\n")
+        print("6: Advanced RAG - HANA Self Query")
```
Comment (EvgeniiSkrebtcov) [inline scripts/step03_explore_examples/02_advanced_rag/main.py:40]: Please align the numbering with: https://github.com/SAP-samples/btp-genai-starter-kit/pull/92.

**Review by EvgeniiSkrebtcov — CHANGES_REQUESTED** [review]

**Comment by prashantrakheja** [comment]
> As far as I can see, this example is not covered in the readme. Is it a reimplementation of a [Self-Query example](https://github.com/SAP-samples/btp-genai-starter-kit/blob/main/scripts/step03_explore_examples/02_advanced_rag/README.md#example-self-query) or something else?

Hi @EvgeniiSkrebtcov,

Thank you, I have now added documentation for this example in README. 

This is a reimplementation of the Self Query example, however it uses the SelfQueryRetriever to perform self-query, whereas the previous example uses a different mechanism (passing filters down to a RetrievalQA chain) to achieve the same thing.

File: scripts/step03_explore_examples/02_advanced_rag/main.py (hunk 1 above)
Comment (prashantrakheja) [reply scripts/step03_explore_examples/02_advanced_rag/main.py:40]: Sure, we will align cc: @sreanikdk

**Review by EvgeniiSkrebtcov — APPROVED** [review]
lgtm

**Pull request merged**